src/test/java/teammates/test/cases/webapi/UpdateInstructorPrivilegeActionTest.java

```diff
@@ -0,0 +1,444 @@
+
+        InstructorPrivilegeData response = (InstructorPrivilegeData) result.getOutput();
+        assertFalse(response.isCanModifyCourse());
+        assertFalse(response.isCanModifySession());
+        assertFalse(response.isCanModifyStudent());
+        assertFalse(response.isCanModifyInstructor());
+
+        assertFalse(response.isCanViewStudentInSections());
+
+        assertFalse(response.isCanSubmitSessionInSections());
+        assertFalse(response.isCanViewSessionInSections());
+        assertFalse(response.isCanModifySessionCommentsInSections());
+
+        // verify the privilege has indeed been updated
+        InstructorAttributes instructor = logic.getInstructorForGoogleId(
+                instructor1OfCourse1.getCourseId(), instructor1OfCourse1.getGoogleId());
+
+        assertFalse(instructor.privileges.isAllowedForPrivilege(
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_COURSE));
+        assertFalse(instructor.privileges.isAllowedForPrivilege(
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION));
+        assertFalse(instructor.privileges.isAllowedForPrivilege(
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR));
+        assertFalse(instructor.privileges.isAllowedForPrivilege(
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT));
+        assertFalse(instructor.privileges.isAllowedForPrivilege(
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS));
+        assertFalse(instructor.privileges.isAllowedForPrivilege(
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_SUBMIT_SESSION_IN_SECTIONS));
+        assertFalse(instructor.privileges.isAllowedForPrivilege(
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_SESSION_IN_SECTIONS));
+        assertFalse(instructor.privileges.isAllowedForPrivilege(
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION_COMMENT_IN_SECTIONS));
+    }
+
+    @Test
+    protected void testExecute_withCourseIdAndSectionName_shouldUpdatePrivilegesInSectionLevel() {
+        InstructorAttributes helper1OfCourse1 = typicalBundle.instructors.get("helperOfCourse1");
+
+        assertFalse(helper1OfCourse1.privileges.isAllowedForPrivilege(
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_COURSE));
+        assertFalse(helper1OfCourse1.privileges.isAllowedForPrivilege(
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION));
+        assertFalse(helper1OfCourse1.privileges.isAllowedForPrivilege(
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR));
+        assertFalse(helper1OfCourse1.privileges.isAllowedForPrivilege(
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT));
+        assertFalse(helper1OfCourse1.privileges.isAllowedForPrivilege(
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS));
```

Comment:
I think you should verify the privilege in **section** level rather than course level in the pre-condition check because the test case tests for section level update rather than course level update.